internal/build/templates/config.sh.tpl

```diff
@@ -8,6 +8,32 @@ useradd -m {{ .Username }} || true
 echo '{{ .Username }}:{{ .Password }}' | chpasswd
 {{ end }}
 
+{{- if .NetworkScript }}
+# Configuring network
+
+cat <<- EOF > /etc/systemd/system/first-boot-network.service
+[Unit]
+Description=Configure network on first boot
+Before=first-boot-complete.target
+Wants=first-boot-complete.target
+
+Before=rke2-server.service rke2-agent.service
```

Comment:
If I understand https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html correctly, before means 

>After= ensures the opposite, that the listed unit is fully started up before the configured unit is started

as it is `type=oneshot` that means immediately after the network.sh has finished running. however, I think at that point nm hasn't finished reconfiguring the network, has it? which means rke2 is potentially running already while network is being reconfigured?

Reply:
>  If unit foo.service contains the setting Before=bar.service and both units are being started, bar.service's start-up is delayed until foo.service has finished starting up. 

While clearly not sufficient, the network configuration script finishes execution very fast and I wasn't able to observe RKE2 starting during its execution, hence I never ran into issues. Obviously, this is not a guarantee it will always work but it will have to be addressed at a later point as discussed earlier.

Reply:
We absolutely need to make sure that we can guarantee that network setup runs before RKE2 starts. The moment that certs are generated and etcd members are established on an incorrect setup (e.g. IP addresses received by DHCP), we cannot easily change them.